File: nuxt_side/test/dot_sfen.spec.js
import { DotSfen } from "@/components/models/dot_sfen.js"

describe("DotSfen", () => {
  test("escape", () => {
    expect(DotSfen.escape("position sfen 9/9/9/9/9/9/9/9/9 b - 1")).toEqual("position.sfen.9/9/9/9/9/9/9/9/9.b.-.1")
    expect(DotSfen.escape("foo")).toEqual("foo")
  })

  test("unescape", () => {
    expect(DotSfen.unescape("position.sfen.9/9/9/9/9/9/9/9/9.b.-.1")).toEqual("position sfen 9/9/9/9/9/9/9/9/9 b - 1")
    expect(DotSfen.unescape("foo")).toEqual("foo")
  })
})

File: spec/models/dot_sfen_spec.rb
require "rails_helper"

RSpec.describe DotSfen, type: :model do
  it "escape" do
    assert { DotSfen.escape("position sfen 9/9/9/9/9/9/9/9/9 b - 1") == "position.sfen.9/9/9/9/9/9/9/9/9.b.-.1" }
    assert { DotSfen.escape("foo") == "foo" }
  end
  it "unescape" do
    assert { DotSfen.unescape("position.sfen.9/9/9/9/9/9/9/9/9.b.-.1") == "position sfen 9/9/9/9/9/9/9/9/9 b - 1" }
    assert { DotSfen.unescape("foo") == "foo" }
  end
end
